charts/tekton-pipelines/values.yaml

```diff
@@ -184,18 +184,30 @@ buildpacks:
       # @default -- see values.yaml for the default values of it
       buildTaskSteps:
         - name: copy-requirements
-          image: docker.io/library/bash:5.1.8
+          image: docker.io/fnndsc/python-poetry:1.5.1
           imagePullPolicy: IfNotPresent
           resources: {}
           workingDir: $(resources.inputs.app.path)
           script: |
             #!/usr/bin/env bash
-            set -e
-            [ -f manage.py ] && (cp requirements/$(params.environment)*.txt requirements.txt \
-              || echo "Unable to find requirements for $(params.environment)")
-            [ -f requirements.txt ] && echo "Found requirements.txt" \
-              || (echo "requirements.txt is not found! Halting execution."; exit 1;)
-
+            set -eo pipefail
+            if [[ -f poetry.lock ]]; then
+              echo "poetry.lock is found, preparing requirements.txt"
+              TARGET="main"
+              if [[ "$(params.environment)" =~ "^dev" ]]; then
+                TARGET="${TARGET},dev"
+              fi
+              poetry export --with ${TARGET} --output requirements.txt
+            elif [[ -f manage.py ]]; then
```

Comment:
1. we should not rely on existance on manage.py
2. you should check existance of `requirements/$(params.environment)*.txt` here
3. no need to use elif - just if.
4. each each should have exit 0
5. last command of this entire script should be exist with non-zero exit code and message that we didn't find any python configuration for installable dependencies via pip or poetry.

Reply:
@dmitry-mightydevops fixed